readthedocs/search/api/v2/serializers.py

```diff
@@ -76,6 +78,28 @@ class PageSearchSerializer(serializers.Serializer):
     highlights = PageHighlightSerializer(source="meta.highlight", default=dict)
     blocks = serializers.SerializerMethodField()
 
+    def __init__(self, *args, projects=None, **kwargs):
+        if projects:
+            context = kwargs.setdefault("context", {})
+            context["projects_data"] = {
+                project.slug: self._build_project_data(project, version.slug)
```

Comment:
Oh, now I remember why this was changed, I was about to use it in this same class, to avoid duplicating the logic at https://github.com/readthedocs/readthedocs.org/blob/2858e6e28619f8d6835b781b3c848824bfd9b3b9/readthedocs/search/api/v2/serializers.py#L117-L132